dist/dfu.js

```diff
@@ -382,7 +382,8 @@
                         case OPCODE.VALIDATE_FIRMWARE:
                             log('complete, reset...');
 
-                            controlChar.service.device.addEventListener("gattserverdisconnected", function() {
+                            server.device.addEventListener("gattserverdisconnected", function() {
```

Comment:
Although in the spec it says BluetoothRemoteGATTCharacteristic contains readonly attribute BluetoothRemoteGATTService service; this isn't the case yet..

Maybe its not implemented or maybe its a bug. But as it is now this will return an error because it is trying to read property 'device' of 'undefined' and the active and reset op code will never be sent resulting in the transfer failing.